src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -487,4 +487,25 @@ public void testBlockCompressionExtension(final String testURIString, final bool
     public void testBlockCompressionExtensionStringVersion(final String testURIString, final boolean expected) {
         Assert.assertEquals(IOUtil.hasBlockCompressedExtension(testURIString), expected);
     }
+
+    @DataProvider
+    public static Object[][] blockCompressedFiles() {
+        return new Object[][]{
+                {TEST_DATA_DIR.resolve("ipsum.txt"), true, false},
+                {TEST_DATA_DIR.resolve("ipsum.txt"), false, false},
+                {TEST_DATA_DIR.resolve("ipsum.txt.gz"), true, false},
+                {TEST_DATA_DIR.resolve("ipsum.txt.gz"), false, false},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz"), true, true},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz"), false, true},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz.wrongextension"), true, false},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgz.wrongextension"), false, true},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgzipped_with_gzextension.gz"), true, true},
+                {TEST_DATA_DIR.resolve("ipsum.txt.bgzipped_with_gzextension.gz"), false, true}

```

Comment:
Nice and complete test cases! Maybe you can add a test with a BAM/BCF file in the repo as an extra to ensure that they are detected as bgz only if there is no check of the extension, but still detected otherwise...

Reply:
good idea, added a bam